org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProvider.java

```diff
@@ -599,5 +617,75 @@ private void applyToNode(AbstractElement node, String actionText, MultiTextEdit
 					}
 				});
 	}
+	
+	@Fix(INVALID_TERMINAL_FRAGMENT_RULE_REFERENCE)
+	public void fixConvertTerminalFragmentToTerminalRule(Issue issue, IssueResolutionAcceptor acceptor) {
+		acceptor.accept(issue, //
+				"Convert terminal fragment to terminal rule", //
+				"Convert terminal fragment to terminal rule", //
+				NULL_QUICKFIX_IMAGE, //
+				(context) -> {
+					IXtextDocument xtextDocument = context.getXtextDocument();
+					xtextDocument.modify(new IUnitOfWork.Void<XtextResource>() {
+
+						@Override
+						public void process(XtextResource state) throws Exception {
+							if (state == null) {
+								return;
+							}
+
+							RuleCall ruleCall = (RuleCall) state.getEObject(issue.getUriToProblem().fragment());
+							ICompositeNode ruleNode = NodeModelUtils.findActualNodeFor(ruleCall.getRule());
+							for (INode node : ruleNode.getAsTreeIterable()) {
+								if (node.getGrammarElement() != null && node.getGrammarElement() == grammarAccess.getTerminalRuleAccess()
+										.getFragmentFragmentKeyword_2_0_0_0()) {
+									ITextRegion fragmentRegion = node.getTextRegion();
+									xtextDocument.replace(fragmentRegion.getOffset(), fragmentRegion.getLength() + 1, "");
```

Comment:
I know that it is extremely unlikely to have that in production, but could you please add a test with this rule declaration:
```
terminal fragment/*comment*/FRAG: 'a';
```

Reply:
I had to adapt the code after introducing the test for the above rule declaration. Could you please verify the changes.